src/genesys/utils.py

```diff
@@ -21,6 +22,9 @@
 from huggingface_hub import snapshot_download
 from datasets import load_dataset
 
+if TYPE_CHECKING:
+    from pathlib import Path
```

Comment:
its fine to import pathlib even if you just use as type hint

Reply:
but the startup will be a few nanoseconds slower and ppl will die!